### ISSUE TITLE
`torch.utils.checkpoint` with BatchNorm

### ISSUE THREAD

**Issue by guozhiyao** [pytorch/pytorch#62471]
## ❓ Questions and Help

When I use the `checkpoint` to warp a module with BatchNorm, will it change the running_average of BN?

### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

**Comment by ezyang** [comment]
I don't see why it wouldn't.

**Comment by guozhiyao** [comment]
@ezyang 
I use a toy model to test the value
``` python
import torch
import torch.nn as nn
from torch.utils.checkpoint import checkpoint
from torch import optim

class bn_linear(nn.Module):
    def __init__(self):
        super(bn_linear, self).__init__()
        self.bn = nn.BatchNorm1d(10, momentum=0.001)
        self.linear = nn.Linear(10, 1)
    def forward(self, x):
        return self.linear(self.bn(x))

class Model(nn.Module):
    def __init__(self):
        super(Model, self).__init__()
        self.s = bn_linear()
    def forward(self, inp):
        return checkpoint(self.s, inp)
        # return self.s(inp)

model = Model()

for i in range(1000):
    x = torch.ones(2,10)*i
    x.requires_grad_(True)
    loss = torch.mean(model(x))

print({n:p for n, p in model.named_buffers()})
```
But the result shows the same.
